x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/notify_badge.test.tsx

```diff
@@ -113,6 +113,23 @@ describe('RulesListNotifyBadge', () => {
     expect(onRuleChanged).toHaveBeenCalled();
   });
 
+  it('should not allow the user to snooze the rule unless they have edit permissions', async () => {
+    render(
+      <RulesListNotifyBadge
+        snoozeSettings={{
+          name: 'rule 1',
+          muteAll: false,
+          isSnoozedUntil: null,
+        }}
+        onRuleChanged={onRuleChanged}
+        snoozeRule={snoozeRule}
+        unsnoozeRule={unsnoozeRule}
+        isRuleEditable={false}
+      />
+    );
+    expect(screen.queryByTestId('rulesListNotifyBadge-unsnoozed')).not.toBeInTheDocument();
```

Comment:
I think you need to check for `rulesListNotifyBadge` as the `buttonWithToolTip` will be undefined because the `button` will be `null`, right?